CHANGELOG.md

```diff
@@ -1,3 +1,514 @@
+
+#### Features
+
+- Add label `lokomotive.kinvolk.io/name: <namespace_name>` to all namespaces ([#646](https://github.com/kinvolk/lokomotive/pull/646)).
+
+- Add admission webhook to lokomotive which disables automount of `default` service account token ([#704](https://github.com/kinvolk/lokomotive/pull/704)).
+
+- [Breaking Change] Kubelet joins cluster using TLS Bootstrapping now, add flag `enable_tls_bootstrap = false` to disable. ([#618](https://github.com/kinvolk/lokomotive/pull/618)).
+
+- Add `csi_plugin_node_selector` and `csi_plugin_toleration` for rook-ceph's CSI plugin ([#892](https://github.com/kinvolk/lokomotive/pull/892)).
+
+
+#### Docs
+
+- Setting up third party OAuth for Grafana ([#542](https://github.com/kinvolk/lokomotive/pull/542)).
+
+- Upgrading bootstrap kubelet ([#592](https://github.com/kinvolk/lokomotive/pull/592)).
+
+- Upgrading etcd ([#802](https://github.com/kinvolk/lokomotive/pull/802)).
+
+- How to add custom monitoring resources? ([#554](https://github.com/kinvolk/lokomotive/pull/554)).
+
+- Kubernetes storage with Rook Ceph on Packet cloud ([#494](https://github.com/kinvolk/lokomotive/pull/494)).
+
+
+#### Bugfixes
+
+- aws: Add check for multiple worker pools with same LB ports ([#889](https://github.com/kinvolk/lokomotive/pull/889)).
+
+- packet: ignore changes to plan and user_data on controller nodes ([#907](https://github.com/kinvolk/lokomotive/pull/907)).
+
+- Introduce platform.PostApplyHook interface and implement it for AKS cluster ([#886](https://github.com/kinvolk/lokomotive/pull/886)).
+
+- aws-ebs-csi-driver: add NetworkPolicy allowing access to metadata ([#865](https://github.com/kinvolk/lokomotive/pull/865)).
+
+- pkg/components/cluster-autoscaler: fix checking device uniqueness ([#768](https://github.com/kinvolk/lokomotive/pull/768)).
+
+#### Development
+
+- Replace use of github.com/pkg/errors.Wrapf with fmt ([#831](https://github.com/kinvolk/lokomotive/pull/831), [#877](https://github.com/kinvolk/lokomotive/pull/877)).
+
+- Refactor assets handling ([#807](https://github.com/kinvolk/lokomotive/pull/807)).
+
+- cli/cmd: improve --kubeconfig-file flag help message formatting ([#818](https://github.com/kinvolk/lokomotive/pull/818)).
+
+- Use host's /etc/hosts entries for bootkube ([#409](https://github.com/kinvolk/lokomotive/pull/409)).
+
+- Refactor Terraform executor ([#794](https://github.com/kinvolk/lokomotive/pull/794)).
+
+- Pass kubeconfig content around rather than file path ([#631](https://github.com/kinvolk/lokomotive/pull/631)).
+
+
+### Upgrading from v0.3.0
+
+TODOs
+
+- Manual upgrade of the kubelet. Add link to the doc here.
+
+#### Lokoctl Host binary upgrades
+
+##### ct
+
+- Update the `ct` Terraform provider to `v0.6.1`, find the install instructions [here](https://github.com/poseidon/terraform-provider-ct/blob/v0.6.1/README.md#install).
+
+#### Configuration syntax changes
+
+##### Disable TLS Bootstrap
```

Comment:
How is that configuration syntax change? Shouldn't this be placed in the changelog above, where breaking change is mentioned? BTW, did we try to allow graceful upgrades? I think it should be possible if we leave old RBAC rules for the time being?

Reply:
A user should not forget to enable that flag, so it is placed under the general section called "Upgrading from v0.3.0".

Reply:
What do you mean by "graceful upgrades"?

Reply:
Graceful upgrades, I mean cluster won't break after upgrading to v0.4.0, which is currently the case, right?

Reply:
I'd use `####` and move it away from "Configuration syntax changes".

Reply:
I'd remove this `if`: https://github.com/kinvolk/lokomotive/pull/618/commits/29bacb2b026daef284afa94012c16b371d155c9c#diff-a1a676d22e219f2a2ca28b4c8152354fR1 and then see if cluster upgrade works fine.

Reply:
How can the cluster upgrade be called fine if the control plane nodes is recreated?

Reply:
They won't be. They will keep using old certificate (as RBAC rule will be in place still), until user decides to re-create them. Then they will be switched to TLS bootstrapping.

Reply:
> They won't be.

When you say that the nodes won't be recreated is it because of the new change: #907 ?

Should we include that change in this release? How does a user decide they want to recreate the nodes?

Reply:
IMO, that will allow graceful migration to TLS bootstrapping, as new nodes/pools will be using it, so I would include that in the release.

>When you say that the nodes won't be recreated is it because of the new change: #907 ?

Yes.

>How does a user decide they want to recreate the nodes?

That's something we should define.

Reply:
If we don't define the node recreation, should we even make the change that you are suggesting? IMO this change wasn't in the milestone and should not be included.

Reply:
We decided to require the parameter now and re-consider it as part of the next release.